### ISSUE TITLE
Wrong insertion allele?

### ISSUE THREAD

**Issue by bnoyvert** [tjiangHIT/cuteSV#66]
Hi,

I think cuteSV might calls variant alleles in a wrong way for some insertions. I provide an example below. Please would it be possible for you to check if it is a bug or an expected call?

I believe the issue is only present in insertions, like the one shown on the attached IGV screenshot. The reference sequence is repeated and there is a 58 base insertion in about half of the reads. Although the insertion appears at different locations I believe it is the same heterozygous variant, and the difference in location is caused by different ways one can put a variant in repeated sequence. 

CuteSV calls one SV in this region (the first variant in the attached vcf file):

6	102204264	cuteSV.INS.0	C	CAAATTTTAGGTATTGCTTTCCTCCACTGTCTAAAACAGAATCTGGATAATGTATCTT

and it is a 57 base insertion. The problem is that the variant allele from the vcf file above is quite different from the sequence inserted in the reads from which cuteSV called the variant. A few examples of the sequences inserted in reads at different locations (extracted from the bam file using samtools mpileup):

"TACATTTTCCTAATCTACATTTATAATGATTACATTTTATAATATACATTTATAATAGC"
"TTTATAATCTACATTTATAAATGGTTACATTTATAATCTACATTTATAAATGGTTACA" 
"TATAAATGGTTACATTTATAATCTACATTTATAAATGGTTACATTTATAATCTACATT" 
"ATTTATAATCTACATTTATAAATGGTTACATTTATAATCTACATTTATAAATGGTTAC"

I think the allele called by cuteSV cannot be traced to the read insertions above.

I have seen other examples similar to this one, can provide them if needed.

The bam and vcf files are attached.
The vcf was generated by this command:
cuteSV example.bam human_g1k_v37.fasta example.vcf ./ --genotype 
The fasta file is the standard 1000 genomes hg19 reference file.
The version of software is cuteSV 1.0.12 cloned from this repository on 12 January 2022.

Many thanks,
Boris

![example](https://user-images.githubusercontent.com/97630376/149393101-9bcecf8e-1377-42ef-a6e9-6acf927d5551.png)
[example.zip](https://github.com/tjiangHIT/cuteSV/files/7865327/example.zip)

**Comment by tjiangHIT** [comment]
Hello @bnoyvert,

Thanks for using cuteSV and reporting this.
cuteSV determines the breakpoint for insertion through the collection of supporting reads and computation of the mean breakpoint from them. After that, cuteSV selects a read randomly and extracts the corresponding length of insertion as the reported insertion sequence.
In the case you are provided, there are 30 reads that contain similarly inserted contig on the close coordination on the reference, so cuteSV reports a 58 bp insertion which alternative allele is _TAAATTTTAGGTATTGCTTTCCTCCACTGTCTAAAACAGAATCTGGATAATGTATCTT_.

In order to figure out the similarity of the alternative allele that cuteSV reported and the consensus of these actually inserted contigs (which is _TATAAATGGTTACATTTATAATCTACATTTATAAATGGTTACATTTATAATCTACATTTATAAATGGTTACATTTA_), we perform SPOA to compare these two sequences. The results are shown below:
![3866172de448073122549a2afc56bbd](https://user-images.githubusercontent.com/7637156/149601483-11532bdd-166e-4f80-92f1-4d5736ae9f04.png)

In conclusion, the two sequences have comparatively high similarity, which indicates the random extraction of the insertion sequence is reliable.

Best,
Tao

**Comment by bnoyvert** [comment]
Hi @tjiangHIT ,

Thank you for looking into the issue and confirming that this is an expected behavior of cuteSV.

The two sequences do indeed have some similarity, but in my opinion it is quite low - the longest stretch of matching bases is just 4 base long.

Many thanks,
Boris

**Comment by tjiangHIT** [comment]
Hello @bnoyvert,

We regenerate the local consensus through the reads including this insertion by adPOA and aligned it to the reference by ngmlr. After that, we compute the similarity between the insertion detected and the allele cuteSV reported, the result is shown below:
![微信图片_20220126174118](https://user-images.githubusercontent.com/7637156/151139667-e69f72a6-9c94-4c48-a850-4bbb21e129f1.png)
These two sequences have relative high similarity and higher than the last time shown one.
We believe this randomly reporting for insertion allele is promising.

Best,
Tao

**Comment by tjiangHIT** [comment]
Hello @bnoyvert,

Thank you so much for your contribution (ref #73).
It seems to make the insertion sequence more sense.
Now I close this issue. Please reopen it if necessary.

Best regards,
Tao
